src/db/oracledb/books-dao.js

```diff
@@ -60,6 +60,63 @@ const getBookById = async (id) => {
   }
 };
 
+/**
+ * Update a specific book by unique ID
+ *
+ * @param {string} id Unique book ID
+ * @param {object} updateData Data to update
+ * @param {object} existingBook Exisitng book to update
+ * @returns {Promise<object>} Promise object represents the updated book or undefined if not found
+ */
+const updateBookById = async (id, updateData, existingBook) => {
+  const connection = await getConnection();
+
+  try {
+    if (!existingBook) {
+      return undefined;
+    }
```

Comment:
Will this piece ever be touched? If `!existingBook`, shouldn't it be caught here? https://github.com/osu-mist/library-api/pull/4/files#diff-c13d6b0e99a9da81a966790595295cd6332bb66bcfc2d6cb3953ec460034a18aR37

Reply:
Even I was thinking the same thing, but let's say when a function calls another function, in C++ language if an interrupt is being called by hardware somehow by any reason, it will still check if it still has an existing resource to change.

Reply:
done.